src/form/Form.jsx

```diff
@@ -700,7 +705,11 @@ function Form({
     updateValues[servar.key] =
       index === null
         ? value
-        : justInsert(fieldValues[servar.key], value, index);
+        : justInsert(
+            fieldValues[servar.key],
+            servar.type === 'file_upload' ? value[0] : value,
```

Comment:
is this basically making the assumption that if index is not null, it's a repeated row so the file field cannot handle multiple files? if so can we add that as a comment

Reply:
the problem was that `value` is an array of length 1 containing the newly updated file. since a field can't be repeating if it allows multiple files, we know it must only have a single value, so can safely access `[0]`. If the file isn't dereferenced, justInsert inserts an array inside the array, ie `[File, [File]]`. It didn't seem appropriate to modify justInsert given the intention of the function

I can add a comment

Reply:
yep ok that's what i thought, a comment would be helpful here!